crates/ide-completion/src/context/analysis.rs

```diff
@@ -510,6 +510,21 @@ fn analyze<'db>(
                     colon_prefix,
                     extern_crate: p.ancestors().find_map(ast::ExternCrate::cast),
                 }
+            } else if p.kind() == SyntaxKind::TOKEN_TREE
+                && p.ancestors().any(|it| ast::Macro::can_cast(it.kind()))
+            {
+                if let Some([_ident, colon, _name, dollar]) = fake_ident_token
+                    .siblings_with_tokens(Direction::Prev)
+                    .filter(|it| !it.kind().is_trivia())
+                    .take(4)
+                    .collect_array()
+                    && dollar.kind() == T![$]
+                    && colon.kind() == T![:]
+                {
+                    CompletionAnalysis::MacroSegment
+                } else {
+                    return None;
+                }
```

Comment:
Can't this re-use the new function in node-ext or is that filtering too aggressively?